### ISSUE TITLE
maximum-line-length for a long link description/title

### ISSUE THREAD

**Issue by jbruggem** [remarkjs/remark-lint#194]
I am faced with a `maximum-line-length` warning for this line, which is part of a list:

```markdown
- [The title of the page I am linking to, which takes 64 chars](sub_folder/other_folder/my_page.md)
```
`maximum-line-length` is set to 80.

Since a new line within a link description is not possible, it seems the warning should not happen here.

Is this a bug ? Or am I looking at this the wrong way ?

**Comment by wooorm** [comment]
You could do it like so:

```markdown
- [The title of the page I am linking to, which takes 64
  chars](sub_folder/other_folder/my_page.md)
```

There is [some code](https://github.com/remarkjs/remark-lint/blob/a82769a0a2bc79ab626c6a12e97a362de2de9ddc/packages/remark-lint-maximum-line-length/index.js#L126-L129) that checks if there’s a possibility of breaking the link, and in this case there is.

**Comment by jbruggem** [comment]
Thank you so much for your quick answer !

It solved my problem. Regrettably, that the syntax coloring for the editors I tried don't support multi-line description for links.

Thanks for your work on remark.

**Comment by jbruggem** [comment]
Addendum :).

I have the same for this line:

```
* [SingleWord](very_long_link)
```

Is that normal as well ?

**Comment by wooorm** [comment]
Hmm, that should work, [and there’s a test for it](https://github.com/remarkjs/remark-lint/blob/a82769a0a2bc79ab626c6a12e97a362de2de9ddc/packages/remark-lint-maximum-line-length/index.js#L28).

What’s the actual code you have?

**Comment by jbruggem** [comment]
@wooorm here's a reproduceable example:

#### when I run remark

```
$ ./node_modules/.bin/remark bug.md
# Hello

-   item 1
-   [SoftwareIntegration](https://very-long-domain.eu/some/thing/a/link/about/integration/that/reaches/)
-   item 3
bug.md
  4:103  warning  Line must be at most 80 characters  maximum-line-length  remark-lint

⚠ 1 warning
```

#### contents bug.md

```
# Hello

- item 1
- [SoftwareIntegration](https://very-long-domain.eu/some/thing/a/link/about/integration/that/reaches/)
- item 3
```
#### remark's config

```json
  "remarkConfig": {
    "plugins": [
      "remark-preset-lint-recommended",
      ["remark-lint-no-html", false],
      ["remark-lint-list-item-indent", "space"],
      ["remark-lint-list-item-spacing", false],
      ["remark-lint-unordered-list-marker-style", "consistent"],
      ["remark-lint-maximum-line-length", 80],
      ["remark-lint-maximum-heading-length", false]
    ]
  }
```

**Comment by jbruggem** [comment]
In relation to my original question: adding a line break in a link title/description as you [recommended](https://github.com/remarkjs/remark-lint/issues/194#issuecomment-445205144) breaks all the other markdown-related tools I use (spell checker, editor syntax highlighting, etc). It might not be a commonly-used pattern. 

For now, I've opted to use another linter (`markdownlint`) for the line-length rule (and still use remark for all the others).

Thank you for sharing this great tool with the community !

**Comment by wooorm** [comment]
I tested this myself and it’s working the way it’s supposed to. Maybe you’re not on the latest versions of dependencies?

<details>

##### `ex.md`

```markdown
# Hello

- item 1
- [SoftwareIntegration](https://very-long-domain.eu/some/thing/a/link/about/integration/that/reaches/)
- item 3
```

##### `config.json`

```json
{
  "plugins": {
    "./packages/remark-lint-maximum-line-length": 80
  }
}
```

##### Script

(running from this repo)

```sh
./node_modules/.bin/remark ex.md --rc-path config.json
```

##### Output

```
# Hello

-   item 1
-   [SoftwareIntegration](https://very-long-domain.eu/some/thing/a/link/about/integration/that/reaches/)
-   item 3
ex.md: no issues found
```

</details>
